### PR TITLE
 Add more assembly version testing for SharedFx/Targeting pack

### PR DESCRIPTION
**Pull request by wtgodbe**
Fixes https://github.com/dotnet/aspnetcore/issues/24541

Adds tests to confirm that assemblies contained in the SharedFx/Targeting pack have versions ending in `0.0`, and only reference things with versions ending in `.0`

One of these tests currently fails - today, the `System.Security.Cryptography.Xml` that we have in the SharedFx (which is versioned `5.0.0.0`) references `System.Security.Cryptography.Pkcs` at `4.0.3.2`, and `System.Memory` at `4.0.1.1`. I believe those are both 3.1 versions. @ericstj @safern @Anipik @joperezr is that expected?

### CODE REVIEW

**Comment by wtgodbe** [comment]
I guess so:
https://github.com/dotnet/runtime/blob/347be9ce2a8a6c5595def6a81229cad976697538/src/libraries/System.Security.Cryptography.Xml/src/System.Security.Cryptography.Xml.csproj#L90-L94

https://github.com/dotnet/runtime/blob/fcd8dd9d41d696341a2bc96cfcaaedf7ff8be3b0/eng/Versions.props#L96

**Comment by dougbu** [comment]
> Adds tests to confirm that assemblies contained in the SharedFx/Targeting pack have versions ending in `0.0`, and only reference things with versions ending in `.0`

Not a big surprise that dotnet/runtime bits have slightly different versioning. Much less worried about our sharedFx dependencies than what shows up in the targeting pack. So, sure, add an exception. And, if  @ericstj @safern @Anipik @joperezr expect similar versioning for 5.0 or 6.0 implementation assemblies, make the exception broader -- perhaps even to all referenced `System.*` assemblies (though that seems a shame).

File: src/Framework/test/SharedFxTests.cs [hunk 1]
```diff
@@ -155,6 +159,48 @@ public void ItContainsValidDepsJson()
...
+        public void SharedFrameworkAssembliesHaveExpectedAssemblyVersions()
+        {
+            // Only test managed assemblies
+            IEnumerable<string> dlls = Directory.GetFiles(_sharedFxRoot, "*.dll", SearchOption.AllDirectories).Where(i => !i.Contains("aspnetcorev2_inprocess"));
+            Assert.NotEmpty(dlls);
+
+            Assert.All(dlls, path =>
+            {
+                var fileName = Path.GetFileNameWithoutExtension(path);
+                var assemblyName = AssemblyName.GetAssemblyName(path);
```
Comment (dougbu) [inline src/Framework/test/SharedFxTests.cs:172]: Where are these two `var`s used❔ Remove if I'm reading this correctly.

**Review by dougbu — APPROVED** [review]
Looks fine pending input from dotnet/runtime and getting the CI to pass

File: src/Framework/test/SharedFxTests.cs (hunk 1 above)
Comment (wtgodbe) [reply src/Framework/test/SharedFxTests.cs]: Nowhere, vestigial from copying stuff from a different test

**Pull request merged**